### ISSUE TITLE
Add support for effect parameters [browser mode]

### ISSUE THREAD

**Issue by qchenevier** [scribbletune/scribbletune#108]
Currently, to manage Tone instruments with scribble, we can either pass a string so that scribble creates the `Tone.Instrument` object itself, or we can pass directly a `Tone.Instrument` object to keep access to it: this enables modifying the parameters during playing.

Example:
- Create object:
```javascript
synth = new Tone.Synth();
scribble
  .clip({ instrument: synth, pattern: 'xx[xx]', notes: 'C3 Cm-3' })
  .start();
```
- Modify its behavior during playing
```javascript
synth.set({oscillator: {type: "sawtooth"}});
```

This is currently not possible if for the `effect` parameter. We cannot pass a `Tone.Effect` object. Scribble support only string for this parameter and will create a `Tone.Effect` instance with the default parameters, that we are not able to customize afterwards.

**Comment by qchenevier** [comment]
@walmik what do you think of implementing a "clever" parameter handling:
- if `effect` is a string, create the `Tone.Effect`
- if `effect` is not a string, assume it is a `Tone.Effect` object to be connected as is to the instrument

**Comment by qchenevier** [comment]
This could also a way to simplify the instrument parameters management. Currently we have: 
- `synth` to pass a string so that scribble creates a `Tone.Instrument`
- `instrument` to pass a `Tone.Instrument` object

It would be simpler to have a single parameter (e.g: `instrument`), accepting either string or `Tone.Instrument` object.

**Comment by qchenevier** [comment]
I can build a PR pretty quickly, as soon as we agree on a design.

**Comment by walmik** [comment]
Hi @qchenevier 

I had created the `effects` param as an array of strings to be passed to Scribbletune only to simplify "chaining" of the effects in a declarative fashion. Coz if you try to add a bunch of effects in Tone, it can be a bit cumbersome. Also from readability point of view, I d struggle to quickly identify the "order" of the effects. Hence this array input was created. So that the order in which you pass the effects, remains clear.

Having said that, I agree with you that the effects are not editable. What if we deployed your idea to cleverly identify string vs Tone Effect? For example, we could accept string as well as Tone Effect objects and make sure to chain them up in the order they were passed via the `effects` array.

I also like the idea of minimizing the api to a single `instrument` param as well. It will need a major or minor version bump depending on the way you choose to implement. If you maintain backward compatibility (by cleverly detecting if `string` or `instanceof` for the `instrument` while leaving `synth` as is) then minor version bump shud be fine.

At some point we should discuss a major version bump (v4) where we can make some assumptions and introduce breaking changes ;)
